tower-reconnect/src/lib.rs

```diff
@@ -144,9 +142,28 @@ where T: NewService + fmt::Debug,
     }
 }
 
+// Manual impl of Debug for State because we should be able to
+// format a state regardless of whether or not the request type
+// is debug.
```

Comment:
Since the `State` type became generic over requests, we had to add this impl of `Debug` as one could no longer be derived if `R` is not `Debug`, but `R` is not a member of the enum.